src/AutoRest.CSharp/Output/Models/DataPlaneRestClient.cs

```diff
@@ -0,0 +1,227 @@
+                        _nextPageMethods.Add(serviceRequest, nextMethod);
+                    }
+                }
+            }
+
+            return _nextPageMethods;
+        }
+
+        private static RestClientMethod BuildNextPageMethod(RestClientMethod method, Operation operation)
+        {
+            var nextPageUrlParameter = new Parameter(
+                "nextLink",
+                "The URL to the next page of results.",
+                typeof(string),
+                defaultValue: null,
+                validateNotNull: true);
+
+            PathSegment[] pathSegments = method.Request.PathSegments
+                .Where(ps => ps.IsRaw)
+                .Append(new PathSegment(nextPageUrlParameter, false, SerializationFormat.Default, isRaw: true))
+                .ToArray();
+            var request = new Request(
+                RequestMethod.Get,
+                pathSegments,
+                Array.Empty<QueryParameter>(),
+                method.Request.Headers,
+                null
+            );
+
+            Parameter[] parameters = method.Parameters.Where(p => p.Name != nextPageUrlParameter.Name)
+                .Prepend(nextPageUrlParameter)
+                .ToArray();
+
+            var responses = method.Responses;
+
+            // We hardcode 200 as expected response code for paged LRO results
+            if (operation.IsLongRunning)
+            {
+                responses = new[]
+                {
+                    new Response(null, new[] { new StatusCodes(200, null) })
+                };
+            }
+
+            return new RestClientMethod(
+                $"{method.Name}NextPage",
+                method.Description,
+                method.ReturnType,
+                request,
+                parameters,
+                responses,
+                method.HeaderModel,
+                bufferResponse: true);
+        }
+
+        public RestClientMethod? GetNextOperationMethod(ServiceRequest request)
+        {
+            EnsureGetNextPageMethods().TryGetValue(request, out RestClientMethod? value);
+            return value;
+        }
+
+        public RestClientMethod GetOperationMethod(ServiceRequest request)
+        {
+            return EnsureNormalMethods()[request];
+        }
+
+        private Parameter BuildClientParameter(RequestParameter requestParameter)
+        {
+            var parameter = BuildParameter(requestParameter);
+            if (requestParameter.Origin == "modelerfour:synthesized/host")
+            {
+                parameter = new Parameter(
+                    "endpoint",
+                    parameter.Description,
+                    typeof(Uri),
+                    parameter.DefaultValue,
+                    parameter.ValidateNotNull
+                );
+            }
+
+            return parameter;
+        }
+
+        public IReadOnlyCollection<Parameter> GetConstructorParameters(CSharpType credentialType, bool includeProtocolOptions = false)
```

Comment:
Doesn't seem this belongs here. The only usage seems to be from the DataPlaneClient and we can inline this implementation there.